crates/static-analysis-kernel/src/analysis/analyze.rs

```diff
@@ -29,6 +29,51 @@ fn get_lines_to_ignore(code: &str, language: &Language) -> LinesToIgnore {
     let mut lines_to_ignore_per_rules: HashMap<u32, Vec<String>> = HashMap::new();
 
     let mut line_number = 1u32;
+    let comment_starters = match language {
+        Language::Python
+        | Language::Starlark
+        | Language::Dockerfile
+        | Language::Elixir
+        | Language::Ruby
+        | Language::Terraform
+        | Language::Yaml
+        | Language::Bash
+        | Language::R => {
+            vec!["#"]
+        }
+        Language::JavaScript | Language::TypeScript | Language::Kotlin | Language::Apex => {
+            vec!["//", "/*"]
+        }
+        Language::Go | Language::Rust | Language::Csharp | Language::Java | Language::Swift => {
+            vec!["//"]
+        }
+        Language::Json => {
+            vec!["impossiblestringtoreach"]
+        }
+        Language::PHP => {
+            vec!["//", "/*", "#"]
+        }
+        Language::Markdown => {
+            vec!["<!-"]
+        }
+        Language::SQL => {
+            vec!["--", "/*"]
+        }
+    };
+
+    // collect the total number of lines that are comments. This way, we can ignore
+    // comment blocks that contains no-dd-sa.
+    let comment_lines = code
+        .lines()
+        .enumerate()
+        .filter(|(_idx, line)| {
+            comment_starters
+                .iter()
+                .any(|m| line.replace(" ", "").starts_with(m))
+        })
+        .map(|(idx, _line)| (idx + 1) as u32)
```

Comment:
For unused variables, just use `_` to indicate we're discarding it.
```suggestion
        .filter(|(_, line)| {
            comment_starters
                .iter()
                .any(|m| line.replace(" ", "").starts_with(m))
        })
        .map(|(idx, _)| (idx + 1) as u32)
```

Or, you could use filter_map

```suggestion
        .filter_map(|(idx, line)| {
            comment_starters
                .iter()
                .any(|m| line.replace(" ", "").starts_with(m))
                .then_some((idx + 1) as u32)
        })
```